// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.staff";
option java_outer_classname = "StaffProto";
option objc_class_prefix = "SM";

package staff;

// The greeting service definition.
service StaffManager {
  // manage staff
  rpc AddDepart (AddDept) returns (AddDeptReply) {}
  rpc ListDoctor (listDoc) returns (listDocReply) {}
  rpc ListDepart (listDept) returns (listDeptReply) {}
}

// The request message containing the user's name.
message AddDoc {
  string EmpID =1;	
  string Qualification=2;
  int32 Fees=3;
  string DeptID=4;
  string Role=5;
  string Name=6;
  string BirthDate=7;
  string Gender=8;
}

message AddDept {	
  string name = 1;
}

// The response message containing the greetings
message AddDeptReply {
  string message = 1;
}

message listDoc{
  string deptId=1;
}

message listDocReply {
  repeated AddDoc message = 1;
}

message listDept {}

message listDeptReply {
  repeated AddDept message = 1;
}